### PR TITLE
QC subworkflow test

### PR DESCRIPTION
**Pull request by alanmmobbs93**
## PR checklist
Resolves [#534](https://github.com/nf-core/rnafusion/issues/534)

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnafusion/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/rnafusion _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`)
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## Added
- Created nf-test for local subworkflow `QC_SUBWORKFLOW`.

## Changed
- Input channels: removed unused `bam_sorted_index` input channel and corrected on main workflow.
- Removed `Channel.empty().mix()` from several channels.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,13 +17,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Add nf-test to local module: `FUSIONREPORT_DOWNLOAD` [#560](https://github.com/nf-core/rnafusion/pull/560)
+- Add nf-test to local subworkflow: `QC_SUBWORKFLOW` [#568](https://github.com/nf-core/rnafusion/pull/568)
 
 ### Changed
 
 - Updated modules and migrated non-specific modules to nf-core/modules [#484](https://github.com/nf-core/rnafusion/pull/484)
 - Updated to nf-core/tools 3.0.2 [#504](https://github.com/nf-core/rnafusion/pull/504)
 - Remove local module `RRNA_TRANSCRIPTS` (replaced by nf-core module) [#541](https://github.com/nf-core/rnafusion/pull/541)
 - Allow fastq files without a dot before .fn(.gz)/.fastq(.gz) files [#548](https://github.com/nf-core/rnafusion/pull/548)
+- Remove unused input channel from `QC_SUBWORKFLOW` [#568](https://github.com/nf-core/rnafusion/pull/568)
```
Comment (nschcolnicov) [inline CHANGELOG.md:28]: Please add this change a single entry, something like "Add nf-tests for QC_SUBWORKFLOW and remove unused input channels for it.

**Review by nschcolnicov — APPROVED** [review]
Very nice!

File: subworkflows/local/qc_workflow/test/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert snapshot(
+                    file( workflow.out.versions[0] ),
+                    file( workflow.out.rnaseq_metrics[0][1] ).readLines()[4..-1],
+                    file( workflow.out.duplicate_metrics[0][1] ).readLines()[4..-1],
+                    file( workflow.out.insertsize_metrics[0][1] ).readLines()[4..-1],
```
Comment (atrigila) [inline subworkflows/local/qc_workflow/test/main.nf.test:123]: I assume the metrics are non-stable, right? I think you can also get the md5 of that piece of stable content.

File: subworkflows/local/qc_workflow/main.nf [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//
+// Check input samplesheet and get read channels
```
[the remark this answers is not in the document]
Comment (alanmmobbs93) [reply subworkflows/local/qc_workflow/main.nf:2]: I didn't see that line, it's not even close to the workflow purpose

File: subworkflows/local/qc_workflow/test/main.nf.test (hunk 2 above)
Comment (alanmmobbs93) [reply subworkflows/local/qc_workflow/test/main.nf.test:123]: Exactly, there's a date + time component there. Let me try again the md5 approach. I couldn't get it working yesterday

**Review by atrigila — APPROVED** [review]
